### PR TITLE
Add static checks workflow (for PHP) and fix phpcs issues

### PR DESCRIPTION
**Pull request by westonruter**
This will catch issues like #199 where PHP 7+ features are being used which would break in PHP 5.6. 

<details><summary>There are a couple dozen failures at present.</summary>

```
FILE: ...ty-experiments/block-interactivity-experiments/wp-directives.php
----------------------------------------------------------------------
FOUND 12 ERRORS AFFECTING 10 LINES
----------------------------------------------------------------------
  11 | ERROR | Missing @package tag in file comment
     |       | (Squiz.Commenting.FileComment.MissingPackageTag)
  27 | ERROR | Mismatched text domain. Expected 'gutenberg' or
     |       | 'default' but got 'wp-directives'.
     |       | (WordPress.WP.I18n.TextDomainMismatch)
Script phpcs --standard=phpcs.xml.dist handling the lint event returned with error code 1
  51 | ERROR | Missing doc comment for function
     |       | wp_directives_loader()
     |       | (Squiz.Commenting.FunctionComment.Missing)
 103 | ERROR | Missing doc comment for function
     |       | wp_directives_add_wp_link_attribute()
     |       | (Squiz.Commenting.FunctionComment.Missing)
 [14](https://github.com/WordPress/block-interactivity-experiments/actions/runs/4547664082/jobs/8017831459?pr=201#step:9:15)3 | ERROR | Missing doc comment for function
     |       | wp_directives_get_client_side_navigation()
     |       | (Squiz.Commenting.FunctionComment.Missing)
 151 | ERROR | Missing doc comment for function
     |       | wp_directives_add_client_side_navigation_meta_tag()
     |       | (Squiz.Commenting.FunctionComment.Missing)
 158 | ERROR | Missing doc comment for function
     |       | wp_directives_client_site_navigation_option()
     |       | (Squiz.Commenting.FunctionComment.Missing)
 168 | ERROR | Missing doc comment for function
     |       | wp_directives_mark_interactive_blocks()
     |       | (Squiz.Commenting.FunctionComment.Missing)
 195 | ERROR | Doc comment for parameter "$parsed_block" missing
     |       | (Squiz.Commenting.FunctionComment.MissingParamTag)
 195 | ERROR | Doc comment for parameter "$source_block" missing
     |       | (Squiz.Commenting.FunctionComment.MissingParamTag)
 195 | ERROR | Doc comment for parameter "$parent_block" missing
     |       | (Squiz.Commenting.FunctionComment.MissingParamTag)
 215 | ERROR | Missing doc comment for function
     |       | process_directives_in_block()
     |       | (Squiz.Commenting.FunctionComment.Missing)
----------------------------------------------------------------------


FILE: ...riments/block-interactivity-experiments/src/directives/utils.php
----------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
  1 | ERROR | Missing file doc comment
    |       | (Squiz.Commenting.FileComment.Missing)
  5 | ERROR | Missing doc comment for function wp_store()
    |       | (Squiz.Commenting.FunctionComment.Missing)
 10 | ERROR | You must use "/**" style comments for a function
    |       | comment
    |       | (Squiz.Commenting.FunctionComment.WrongStyle)
 10 | ERROR | 'string' type declaration is not present in PHP version
    |       | 5.6 or earlier
    |       | (PHPCompatibility.FunctionDeclarations.NewParamTypeDeclarations.stringFound)
 27 | ERROR | Missing doc comment for function set_style()
    |       | (Squiz.Commenting.FunctionComment.Missing)
----------------------------------------------------------------------


FILE: ...teractivity-experiments/src/directives/wp-process-directives.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 1 | ERROR | Missing file doc comment
   |       | (Squiz.Commenting.FileComment.Missing)
 6 | ERROR | Missing doc comment for function
   |       | wp_process_directives()
   |       | (Squiz.Commenting.FunctionComment.Missing)
----------------------------------------------------------------------


FILE: ...vity-experiments/src/directives/class-wp-directive-processor.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 1 | ERROR | Missing file doc comment
   |       | (Squiz.Commenting.FileComment.Missing)
 3 | ERROR | Missing doc comment for class WP_Directive_Processor
   |       | (Squiz.Commenting.ClassComment.Missing)
----------------------------------------------------------------------


FILE: ...activity-experiments/src/directives/class-wp-directive-store.php
----------------------------------------------------------------------
FOUND 9 ERRORS AFFECTING 8 LINES
----------------------------------------------------------------------
  1 | ERROR | Missing file doc comment
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
  1 | ERROR | Missing file doc comment
    |       | (Squiz.Commenting.FileComment.Missing)
  3 | ERROR | Doc comment short description must start with a capital
    |       | letter
    |       | (Generic.Commenting.DocComment.ShortNotCapital)
 20 | ERROR | Missing doc comment for function
    |       | test_directive_sets_inner_html_based_on_attribute_value_and_retains_html()
    |       | (Squiz.Commenting.FunctionComment.Missing)
 35 | ERROR | Missing doc comment for function
    |       | test_directive_overwrites_inner_html_based_on_attribute_value()
    |       | (Squiz.Commenting.FunctionComment.Missing)
----------------------------------------------------------------------


FILE: ...ctivity-experiments/phpunit/directives/attributes/wp-context.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
  1 | ERROR | Missing file doc comment
    |       | (Squiz.Commenting.FileComment.Missing)
  3 | ERROR | Doc comment short description must start with a capital
    |       | letter
    |       | (Generic.Commenting.DocComment.ShortNotCapital)
 [19](https://github.com/WordPress/block-interactivity-experiments/actions/runs/4547664082/jobs/8017831459?pr=201#step:9:20) | ERROR | Missing doc comment for function
    |       | test_directive_merges_context_correctly_upon_wp_context_attribute_on_opening_tag()
    |       | (Squiz.Commenting.FunctionComment.Missing)
 [42](https://github.com/WordPress/block-interactivity-experiments/actions/runs/4547664082/jobs/8017831459?pr=201#step:9:43) | ERROR | Missing doc comment for function
    |       | test_directive_resets_context_correctly_upon_closing_tag()
    |       | (Squiz.Commenting.FunctionComment.Missing)
----------------------------------------------------------------------
```

</details> 

They are corrected in this PR, except for the issues in PHPUnit test files, which are suppressed for now.

Closes #199.

### CODE REVIEW

**Comment by ockham** [comment]
Thank you, much appreciated!

> Should they be fixed as part of this PR?

I'd say we shouldn't merge this PR with failing tests (as the :x: on all new/rebased PRs would likely confuse people) so we should fix them. We could theoretically do that in a separate PR (that would need to go in first), but it probably makes more sense to do it right here.

I'll start pushing commits to fix things if that's okay 😊 (If you'd rather I didn't, feel free to revert.)

**Comment by westonruter** [comment]
Feel free to push. I'll push first if you don't get to it.

**Comment by westonruter** [comment]
Unit tests are failing due to the `6.2` tag not being pushed to WordPress/WordPress. See https://make.wordpress.org/systems/2023/03/28/github-ssh-host-key-update-svn-git-sync/#comment-2079

**Comment by DAreRodz** [comment]
I re-run the PHP tests, and they're passing now. Thanks, @westonruter for opening this PR. 😊

**Comment by ockham** [comment]
> Feel free to push. I'll push first if you don't get to it.

Beat me to it! Thank you very much 😊

File: src/directives/attributes/wp-style.php [hunk 1]
```diff
@@ -1,7 +1,19 @@
 <?php
+/**
+ * Process wp-html directive attribute.
```
Comment (ockham) [inline src/directives/attributes/wp-style.php]:
```suggestion
 * Process wp-style directive attribute.
```

File: src/directives/attributes/wp-text.php [hunk 2]
```diff
@@ -1,7 +1,19 @@
 <?php
+/**
+ * Process wp-html directive attribute.
```
Comment (ockham) [inline src/directives/attributes/wp-text.php]:
```suggestion
 * Process wp-text directive attribute.
```

File: src/directives/utils.php [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
 function wp_store( $data ) {
 	WP_Directive_Store::merge_data( $data );
 }
 
-// TODO: Implement evaluation of complex logical expressions.
-function evaluate( string $path, array $context = array() ) {
+/**
+ * Obtain data from store and context from the provided path.
+ *
+ * TODO: Implement evaluation of complex logical expressions.
```
Comment (ockham) [inline src/directives/utils.php]: Could use PHPDoc `@todo`

```suggestion
 * @todo Implement evaluation of complex logical expressions.
```

File: src/directives/wp-html.php [hunk 4]
```diff
@@ -1,4 +1,9 @@
 <?php
+/**
+ * Polyfill WP_HTML_Directives.
```
Comment (ockham) [inline src/directives/wp-html.php]:
```suggestion
 * Polyfill WP_HTML_Tag_Processor.
```

**Review by ockham — COMMENTED** [review]
Left a few notes. LGTM overall though!

**Review by ockham — APPROVED** [review]
LGTM! Thank you very much 🚢

**Comment by westonruter** [comment]
What's the etiquette for merging approved PRs? I have the ability to merge, but I don't want to make assumptions.

**Comment by michalczaplinski** [comment]
I haven't looked at the content of this particular PR, but generally approved == the author is free to merge it. 

So go ahead with the merge! 👍

**Pull request merged**